vowpalwabbit/OjaNewton.cc

```diff
@@ -81,6 +82,15 @@ struct OjaNewton
         {
           float r1 = merand48(all->random_state);
           float r2 = merand48(all->random_state);
+
+          //in the box-muller tranform, r1 should be strictly positive: https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform
+          //redraw until r1 > 0
+          while (r1 < FLOAT_MIN)
```

Comment:
Why not check for zero?

Reply:
In order to avoid numeric instabilities when `0 < r1 < FLOAT_MIN`, especially for the `log(r1)` calculation happening next.

Reply:
That's impossible to happen.
FLOAT_MIN, as you defined, is, by design, the smallest number bigger than zero that can be represented by a float.

Reply:
Ok, thanks. I will check for zero instead.